iosApp/iosApp/feature/scorelist/FilterPanelSwiftUI.swift

```diff
@@ -0,0 +1,268 @@
+                                    slider1.lastStoredOffset = slider1.offset
+                                }
+                        )
+                    
+                    Circle()
+                        .fill(.primary)
+                        .frame(width: 15, height: 15)
+                        .contentShape(.rect)
+                        .offset(x: slider2.offset)
+                        .gesture(
+                            DragGesture(minimumDistance: 0)
+                                .onChanged { value in
+                                    // Calculating offset
+                                    var translation = value.translation.width + slider2.lastStoredOffset
+                                    translation = min(max(translation, slider1.offset + minimumDistance), maxSliderWidth)
+                                    slider2.offset = translation
+                                    calculateNewRange(reader.size)
+                                }.onEnded { _ in
+                                    slider2.lastStoredOffset = slider2.offset
+                                }
+                        )
+                }
+            }
+            .frame(maxHeight: .infinity)
+            .task {
+                guard !isInitial else { return }
+                isInitial = true
+                try? await Task.sleep(for: .seconds(0))
+                let maxWidth = reader.size.width - 30
+                
+                // Converting selection range into offset
+                let start = value.lowerBound.interpolate(inputRange: [range.lowerBound, range.upperBound], outputRange: [0, maxWidth])
+                let end = value.upperBound.interpolate(inputRange: [range.lowerBound, range.upperBound], outputRange: [0, maxWidth])
+                
+                slider1.offset = start
+                slider1.lastStoredOffset = start
+                slider2.offset = end
+                slider2.lastStoredOffset = end
+                calculateNewRange(reader.size)
+            }
+        }
+        .frame(height: 20)
+    }
+    
+    private func calculateNewRange(_ size: CGSize) {
+        indicatorWidth = slider2.offset - slider1.offset
+        
+        let maxWidth = size.width - 30
+        // Calculating new range values
+        let startProgress = slider1.offset / maxWidth
+        let endProgress = slider2.offset / maxWidth
+        
+        // Interpolating between upper and lower bounds
+        let newRangeStart = range.lowerBound.interpolated(towards: range.upperBound, amount: startProgress)
+        let newRangeEnd = range.lowerBound.interpolated(towards: range.upperBound, amount: endProgress)
+        
+        // Update value
+        value = newRangeStart...newRangeEnd
+    }
+    
+    private struct GestureProperties {
+        var offset: CGFloat = 0
+        var lastStoredOffset: CGFloat = 0
+    }
+}
+
+// Interpolation
+extension CGFloat {
```

Comment:
This could potentially be useful in a more common Utils file, but not gonna block on this.